pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -303,41 +303,48 @@ def _on_evaluation_epoch_end(self) -> None:
         self.trainer._logger_connector.on_epoch_end()
 
     @staticmethod
-    def _get_keys(data: dict) -> Iterable[str]:
-        if any(isinstance(v, dict) for v in data.values()):
-            for v in data.values():
-                yield from apply_to_collection(v, dict, dict.keys)
-        else:
-            yield from data.keys()
+    def _get_keys(data: dict) -> Iterable[Tuple[str, ...]]:
+        for k, v in data.items():
+            if isinstance(v, dict):
+                for new_key in apply_to_collection(v, dict, EvaluationLoop._get_keys):
+                    yield k, *new_key
+            else:
+                yield k,
 
     @staticmethod
-    def _find_value(data: dict, target: str) -> Iterable[Any]:
-        for k, v in data.items():
-            if k == target:
-                yield v
-            elif isinstance(v, dict):
-                yield from EvaluationLoop._find_value(v, target)
+    def _find_value(data: dict, target: Iterable[str]) -> Optional[Any]:
+        target_start, *rest = target
+        if target_start not in data:
+            return None
+        result = data[target_start]
+        if not rest:
+            return result
+        return EvaluationLoop._find_value(result, rest)
 
     @staticmethod
     def _print_results(results: List[_OUT_DICT], stage: str) -> None:
         # remove the dl idx suffix
         results = [{k.split("/dataloader_idx_")[0]: v for k, v in result.items()} for result in results]
-        metrics = sorted({k for keys in apply_to_collection(results, dict, EvaluationLoop._get_keys) for k in keys})
-        if not metrics:
+        metrics_paths = {k for keys in apply_to_collection(results, dict, EvaluationLoop._get_keys) for k in keys}
+        if not metrics_paths:
             return
+
+        metrics_strs = [":".join(metric) for metric in metrics_paths]
+        # sort both lists based on metrics_strs
+        metrics_paths, metrics_strs = zip(*sorted(zip(metrics_paths, metrics_strs)))
```

Comment:
Doesn't this sort based on `metric_paths` since it's the first item in the inner zip?

Meaning the comment is wrong.

Note that I find sorting based on `metric_strs` more intuitive

Reply:
You're absolutely right, I don't know how I missed it. I pushed a correct version now